src/corehost/common/pal.unix.cpp

```diff
@@ -83,12 +86,22 @@ bool pal::is_path_rooted(const pal::string_t& path)
 bool pal::get_default_packages_directory(pal::string_t* recv)
 {
     recv->clear();
-    if (!pal::getenv("HOME", recv))
+    pal::string_t dir;
+    if (!pal::getenv("HOME", &dir))
+    {
+        struct passwd* pw = getpwuid(getuid());
+        if (pw && pw->pw_dir)
+        {
+            dir.assign(pw->pw_dir);
+        }
+    }
+    if (dir.empty())
     {
         return false;
     }
-    append_path(&*recv, _X(".nuget"));
-    append_path(&*recv, _X("packages"));
+    append_path(&dir, _X(".nuget"));
+    append_path(&dir, _X("packages"));
+    recv->assign(dir);
```

Comment:
With this change in place, HOME is no longer required to be defined. CI machines do not have it defined and that broke CoreCLR builds when using CLI to perform restore. See https://github.com/dotnet/coreclr/commit/f025f9c678b7194d3b125d0cfa4f8c8e9d27beab#diff-c02be70df3e835d946f675bd0bedc234 for details.

You should file an issue to remove the CI specific change from CoreCLR once CoreCLR uses the updated CLI.

Otherwise, change LGTM